projects/vdk-core/src/vdk/internal/builtin_plugins/ingestion/ingester_base.py

```diff
@@ -439,6 +439,21 @@ def _payload_poster_thread(self):
                     for payload_dict in payload_obj:
                         self.__verify_payload_format(payload_dict=payload_dict)
 
+                    if ingestion_metadata:
+                        updated_dynamic_params: Optional[dict] = ingestion_metadata.pop(
+                            IIngesterPlugin.UPDATED_DYNAMIC_PARAMS, None
+                        )
+                        if updated_dynamic_params:
+                            destination_table = (
+                                updated_dynamic_params.get("destination_table")
+                                or destination_table
+                            )
+                            target = updated_dynamic_params.get("target") or target
+                            collection_id = (
+                                updated_dynamic_params.get("collection_id")
+                                or collection_id
+                            )
+
                     try:
                         ingestion_metadata = self._ingester.ingest_payload(
                             payload=payload_obj,
```

Comment:
why not simplify by reusing `ingestion_metadata` directly, instead of subkey-ing? in case any real parameter name is found in ingestion_metadata, just use it as a parameter for ingest_payload?

Reply:
I believe sub-keying `ingestion_metadata` actually makes it cleaner, as we essentially create a sub-section which if exists, then at least one parameter has been updated updated, and we don't need to have a separate check for each and every parameter.